pkg/driver/driver_aws.go

```diff
@@ -70,6 +70,10 @@ func (d *AWSDriver) Create() (string, string, error) {
 	}
 	metrics.APIRequestCount.With(prometheus.Labels{"provider": "aws", "service": "ecs"}).Inc()
 
+	if len(output.Images) < 1 {
```

Comment:
Wondering if makes sense to add checks for `Spec.BlockDevices` and `Spec.NetworkInterfaces` as well.

Afaik we tackle these cases in the validation package but double-check here should be fine.
- https://github.com/gardener/machine-controller-manager/blob/master/pkg/apis/machine/validation/awsmachineclass.go#L70 
Not sure though how it still panicked.

Reply:
`BlockDevices` and `NetworkInterfaces` are in `spec`. For them, validation is enough.

The AMI check is for driver API call response, hence this check. I could not see any other unchecked indexing of arrays/slices in the `Create` call at least.

Reply:
i don't believe it fixes #278. New comment: https://github.com/gardener/machine-controller-manager/issues/278#issuecomment-501674918
Also, if the image is not found, it should have been caught here:
https://github.com/gardener/machine-controller-manager/pull/279/files#diff-9e1fc8d4d5952f923e158f3a60af4036R67

Reply:
The error returned would be `InvalidAMIID.NotFound` according to https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html#CommonErrors

Reply:
Ideally yes. But we have seen AWS SDK issues (especially, related to error codes) with the CCM as well recently with stale nodes not being cleaned up.

Besides, the logs reported in #278 are about `index out of range` on this exact line. And the image clearly doesn't exist. Not checking length before `output.Images[0]` is not ideal either even if the error codes are consistent.

Reply:
> Not sure though how it still panicked.

@hardikdr This is not the case where `spec.API` is `nil`. It is not `nil` but it does not exist. This case cannot be checked during validation at all.

Also, the SDK or the EC2 API seems to be misbehaving as @ggaurav10 pointed out.

Reply:
According to https://github.com/gardener/machine-controller-manager/issues/278#issuecomment-501674918 the AMI exists. Still panic is seen. With this PR, MCM would not panic, and return "Image not found".

Reply:
> The error returned would be InvalidAMIID.NotFound according to https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html#CommonErrors

I don't think this is entirely correct, now that I think about it. The CommonErrors listing is a list of all common errors. The documentation does not where which API returns which error for what reason?

It makes sense for `RunInstances` to return `InvalidAMIID.NotFound`. It does not make any sense for `DescribeImages` to return `InvalidAMIID.NotFound`. No?

Reply:
According to https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeImages.html
`DescribeImages` should also return common errors if I understood it correct.

Reply:
Tried the scenario with random AMI ID that doesn't exist. Following error is returned by `DescribeImages`:
 ```
InvalidAMIID.NotFound: The image id '[ami-012abdf0d2781f0b5]' does not exist
	status code: 400, request id: fa359937-1b4e-4596-a301-3d3932d5f565
```
The returned `output` struct is empty.
Having said that, there is no harm in still checking of the array `Images` has zero length or not. :)